components/layout/block.rs

```diff
@@ -712,7 +712,7 @@ impl BlockFlow {
 
         // Shift all kids down (or up, if margins are negative) if necessary.
         if block_start_margin_value != Au(0) {
-            for kid in self.base.child_iter() {
+            for kid in self.base.child_iter_mut() {
```

Comment:
Okay, so one of the problems is that this function shouldn't be renamed. This method isn't the same as the one you've changed; the one you've changed has a parameter for Flow.